src/main/java/net/earthcomputer/clientcommands/features/MappingsHelper.java

```diff
@@ -0,0 +1,226 @@
+            .uri(URI.create("https://piston-meta.mojang.com/mc/game/version_manifest_v2.json"))
+            .GET()
+            .timeout(Duration.ofSeconds(5))
+            .build();
+        httpClient.sendAsync(versionsRequest, HttpResponse.BodyHandlers.ofString())
+            .thenApply(HttpResponse::body)
+            .thenAccept(versionsBody -> {
+                JsonObject versionsJson = JsonParser.parseString(versionsBody).getAsJsonObject();
+                String versionUrl = versionsJson.getAsJsonArray("versions").asList().stream()
+                    .map(JsonElement::getAsJsonObject)
+                    .filter(v -> v.get("id").getAsString().equals(version))
+                    .map(v -> v.get("url").getAsString())
+                    .findAny().orElseThrow();
+
+                HttpRequest versionRequest = HttpRequest.newBuilder()
+                    .uri(URI.create(versionUrl))
+                    .GET()
+                    .timeout(Duration.ofSeconds(5))
+                    .build();
+                httpClient.sendAsync(versionRequest, HttpResponse.BodyHandlers.ofString())
+                    .thenApply(HttpResponse::body)
+                    .thenAccept(versionBody -> {
+                        JsonObject versionJson = JsonParser.parseString(versionBody).getAsJsonObject();
+                        String mappingsUrl = versionJson
+                            .getAsJsonObject("downloads")
+                            .getAsJsonObject("client_mappings")
+                            .get("url").getAsString();
+
+                        HttpRequest mappingsRequest = HttpRequest.newBuilder()
+                            .uri(URI.create(mappingsUrl))
+                            .GET()
+                            .timeout(Duration.ofSeconds(5))
+                            .build();
+                        httpClient.sendAsync(mappingsRequest, HttpResponse.BodyHandlers.ofString())
+                            .thenApply(HttpResponse::body)
+                            .thenApply(StringReader::new)
+                            .thenAccept(reader -> {
+                                try {
+                                    MappingReader.read(reader, MappingFormat.PROGUARD_FILE, mojmapOfficial);
+                                } catch (IOException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            });
+                    });
+            });
+
+        try (InputStream stream = FabricLoader.class.getClassLoader().getResourceAsStream("mappings/mappings.tiny")) {
+            if (stream == null) {
+                throw new IOException("Could not find mappings.tiny");
+            }
+            MappingReader.read(new InputStreamReader(stream), IS_DEV_ENV ? MappingFormat.TINY_2_FILE : MappingFormat.TINY_FILE, officialIntermediaryNamed);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Exceptions should be handled gracefully here. What if Fabric loader decides to change where they store the mappings? We don't want to crash the whole game if this one command can't be used.